### PR TITLE
[Serverless] add serverless-specific debug lines for traces and metrics

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

- Adds `log.DebugfServerless` and `log.DebugServerless` to allows extra logging in a serverless context only
- Add serverless debug logs for trace payloads
- Add serverless debug logs for metrics

Screen Shot 2022-01-11 at 11 25 48 AM
Screen Shot 2022-01-11 at 10 55 38 AM

### Motivation

Easier support

### Describe how to test/QA your changes

Covered by unit-tests

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by nhinsch — APPROVED** [review]

File: pkg/util/log/log.go [hunk 1]
```diff
@@ -771,3 +785,8 @@ func ChangeLogLevel(l seelog.LoggerInterface, level string) error {
 	// need to return something, just set to Info (expected default)
 	return errors.New("cannot change loglevel: logger not initialized")
 }
+
+// isServerless returns whether or not the agent is running in a serverless context
+func isServerless() bool {
+	return len(os.Getenv("AWS_LAMBDA_FUNCTION_NAME")) > 0
```
Comment (remeh) [inline pkg/util/log/log.go:791]: You could consider leveraging the `serverless` tag instead here.

Having a `isServerless()` implementation returning `false` when the `serverless` tag has not been set on build etc.

Not a strong opinion.

**Comment by nhinsch** [comment]
Since this PR affects non-serverless files, can we merge it directly into `main` instead of the working branch? Otherwise we will need to get review a second time on the merge PR.

File: pkg/util/log/log.go (hunk 1 above)
Comment (maxday) [reply pkg/util/log/log.go]: Good idea, I've just updated the PR

**Review by remeh — APPROVED** [review]
LGTM, make sure to get a review from agent-apm please 🙇

**Review by piochelepiotr — APPROVED** [review]

**Pull request merged**